src/include/duckdb/planner/binder.hpp

```diff
@@ -174,8 +179,8 @@ class Binder : public std::enable_shared_from_this<Binder> {
 	void BindOnConflictClause(LogicalInsert &insert, TableCatalogEntry &table, InsertStatement &stmt);
 
 	static void BindSchemaOrCatalog(ClientContext &context, string &catalog, string &schema);
-	static void BindLogicalType(ClientContext &context, LogicalType &type, optional_ptr<Catalog> catalog = nullptr,
-	                            const string &schema = INVALID_SCHEMA);
+	void BindLogicalType(LogicalType &type, optional_ptr<Catalog> catalog = nullptr,
```

Comment:
This is no longer a static method because the binding of USER types perform a CatalogEntry lookup which we want to perform the CatalogEntryRetrievers callback on.

In the places that previously used this as a static method without being inside a Binder or having access to a binder I've created a binder on the spot.
In a lot of these places a binder was already created later, so I don't think any harm was done here.